api/src/main/java/org/pmiops/workbench/notebooks/NotebooksConfig.java

```diff
@@ -89,8 +89,8 @@ public ClusterApi serviceClusterApi(@Qualifier(NOTEBOOKS_SERVICE_CLIENT) ApiClie
     return api;
   }
 
-  private void addNotebooksDefaultHeader(ApiClient apiClient) {
+  private void addNotebooksDefaultHeader(ApiClient apiClient, String xAppHeaderValue) {
```

Comment:
Same comment here as above – maybe refactoring as an ApiClient builder method would make more sense?

If you don't think that path is worthwhile, I'd be inclined to remove this method and simply make the apiClient.addDeafultHeader call independently in each of the usage locations. This method feels a bit like needless redirection.